x-pack/legacy/plugins/rollup/server/lib/license_pre_routing_factory/license_pre_routing_factory.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  KibanaRequest,
+  KibanaResponseFactory,
+  RequestHandler,
+  RequestHandlerContext,
+} from 'src/core/server';
+import { PLUGIN } from '../../../common';
+import { LICENSE_STATUS_VALID } from '../../../../../common/constants/license_status';
+import { ServerShim } from '../../types';
+
+export const licensePreRoutingFactory = (
```

Comment:
I wonder if we should first agree on the solution for this or if we should merge this PR and change it later. 
I would think it would be good to first agree on a common pattern (probably a class exposed) to avoid copy/pasting this code + the tests in all our apps.

WDYT @alisonelizabeth @jloleysens @cjcenizal ?

Reply:
I agree with @jloleysens. I'd prefer not to block this PR and address when I finish the migration for rollups and adopt the NP licensing plugin.